[FDORA-29] feat: 교환/반품 api 구현

chakisuk opened the pull request:
## 📌 작업 내용
- 교환/반품 api 구현
- Refund 엔티티 isProcess 컬럼 추가

<br><br>
## 💡 필수확인
```sql
alter table refund add column is_process bit;
```
<br><br>
## 📆 작업기간
2025-04-21 ~ 2025-04-22

<br><br>
## 📷 스크린샷(선택)

<br><br>
## ✅ 참고 사항(선택)

chakisuk committed:
feat: 환불 요청 api 및 RefundType 조회 api 구현

- Refund 엔티티 isProcess 컬럼 추가(판매자의 승인 여부 확인용도)

FDORA-29

chakisuk committed:
chore: 필요 없는 코드 수정 및 결제 정보 추가

FDORA-29

honee8583 commented on src/main/java/com/farmdora/farmdorabuyer/orders/controller/RefundController.java:
```diff
@@ -0,0 +1,48 @@
+                .body(new HttpResponse(HttpStatus.OK, "환불 요청이 성공적으로 등록되었습니다.", response));
+    }
+
+    @GetMapping("order/refundTypes")
```
여기 맨앞에 "/" 없어도 정상작동하나요?

chakisuk replied:
둘 다 정상작동합니다

chakisuk replied:
> 여기 맨앞에 "/" 없어도 정상작동하나요?

둘 다 정상작동합니다

honee8583 commented on src/main/java/com/farmdora/farmdorabuyer/orders/repository/PayRepository.java:
```diff
@@ -12,7 +12,7 @@ public interface PayRepository extends JpaRepository<Pay, Integer> {
 
     // 주문 ID로 결제정보 조회
-    Optional<Pay> findByOrderId(Integer orderId);
+    Pay findByOrderId(Integer orderId);
 
 
     Pay findByOrder(Order order);
```
`Optional`로 리턴하지 않도록 한 이유가 먼가요?

chakisuk replied:
> `Optional`로 리턴하지 않도록 한 이유가 먼가요?

orderId가 있다면 Pay값이 100%있다고 생각해서 바꾸긴 했습니다.
null값이 안 나오지 않을까...

honee8583 commented on src/main/java/com/farmdora/farmdorabuyer/orders/controller/RefundController.java:
```diff
@@ -0,0 +1,48 @@
+
+        Integer userId = 1;
+
+        List<MultipartFile> imageList = images != null ? Arrays.asList(images) : new ArrayList<>();
```
`List`로 변환하지 않아도 배열로 처리할 수 있을듯?

chakisuk replied:
> `List`로 변환하지 않아도 배열로 처리할 수 있을듯?

할 수 있긴한데 컨트롤러에서 배열로 받고, 서비스에서는 List로 처리하는 패턴을 사용하기도 한다고 해서 해봤습니다

chakisuk committed:
fix: RefundController 경로 수정 및 RefundDTO 수정

- controller 경로 "/" 추가
- 교환/반품 확인 작업을 다른 페이지에서 사용하지 않아 RefundListItem DTO 삭제

FDORA-29

chakisuk committed:
chore: ReviewRepository 파일명 오타 수정

FDORA-29
